### PR TITLE
feat:  Additional recipients in generate report

### PR DESCRIPTION
**Pull request by kanikabansal-juspay**
## Type of Change

- [x] Bugfix
- [x] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description

- Addition of multiple email field in the generate report payment operations.
- Report type ( view ) information displayed in the modal box/
- Rendering bug in Pill Input . Earlier on clicking for editing one email makes other emails disappear because of selective rendering.

https://github.com/user-attachments/assets/31a7afbe-bfd0-4fbc-bf6e-9d947d4e0223

## Motivation and Context

Feature request.

## How did you test it?

Attached Video.

## Where to test it?

- [x] INTEG
- [x] SANDBOX
- [ ] PROD

## Checklist

- [x] I ran `npm run re:build`
- [x] I reviewed submitted code
- [ ] I added unit tests for my changes where possible

### CODE REVIEW

**Review by JeevaRamu0104 — APPROVED** [review]

**Review by JeevaRamu0104 — DISMISSED** [review]

File: src/components/form/PillInput.res [hunk 1]
```diff
@@ -117,34 +117,27 @@ let make = (~name, ~initialItems: array<string>=[], ~placeholder, ~duplicateChec
...
-                onInput=handleEditChange
-                onKeyDown={ev => handleEditKeydown(item, ev)}
-                className="rounded-md p-1 flex-grow"
-              />
-              <button onClick={_ => removeItem(item)} className=" hover:bg-gray-300 rounded-md p-1">
-                <Icon name="cross-outline" size=14 />
-              </button>
-            </div>
+        <div key={Int.toString(i)} className="flex flex-wrap gap-1 p-1 border rounded-md">
+          <RenderIf condition={editingItem == Some(item)}>
```
Comment (Riddhiagrawal001) [inline src/components/form/PillInput.res:121]:
```suggestion
          <RenderIf condition={editingItem->Option.isSome}>
```

File: src/components/form/PillInput.res [hunk 2]
```diff
@@ -117,34 +117,27 @@ let make = (~name, ~initialItems: array<string>=[], ~placeholder, ~duplicateChec
...
+              value={editInput}
+              onBlur={_ => saveItem(item)}
+              onInput=handleEditChange
+              onKeyDown={ev => handleEditKeydown(item, ev)}
+              className="rounded-md p-1 flex-grow"
+            />
           </RenderIf>
-        | None =>
-          <div key={Int.toString(i)} className="flex flex-wrap gap-1 p-1 border rounded-md">
+          <RenderIf condition={editingItem != Some(item)}>
```
Comment (Riddhiagrawal001) [inline src/components/form/PillInput.res:131]:
```suggestion
          <RenderIf condition={editingItem->Option.isNone}>
```

**Review by Riddhiagrawal001 — APPROVED** [review]

File: src/components/form/PillInput.res (hunk 1 above)
Comment (kanikabansal-juspay) [reply src/components/form/PillInput.res:121]: here I am not checking if it has some value or not
I am doing direct comparision to check if the current `editingItem` is equal to the specific item

File: src/components/form/PillInput.res (hunk 2 above)
Comment (kanikabansal-juspay) [reply src/components/form/PillInput.res:131]: same here

**Review by JeevaRamu0104 — APPROVED** [review]

**Review by sagarnaikjuspay — APPROVED** [review]

**Pull request merged**